### PR TITLE
Update Istio helm charts from kubeflow v1.7 upstream

### PR DESCRIPTION
**Pull request by jsitu777**
**Description of your changes:**

update istio, cluster-local-gateway, and kubeflow-istio-resource helm chart from istio 1-14 to istio 1-16
update installation config and kubeflow installation/uninstallation script to include the above change

**Testing:**
- [ ] Unit tests pass
- [x] e2e tests pass

### CODE REVIEW

File: tools/helmify/src/config.yaml [hunk 1]
```diff
@@ -72,10 +72,10 @@ oidc-authservice:
 
 cluster-local-gateway:
   kustomization_paths:
-    - upstream/common/istio-1-14/cluster-local-gateway/base
+    - upstream/common/istio-1-16/cluster-local-gateway/base
   output_helm_chart_path: charts/common/cluster-local-gateway
-  version: 0.1.0
-  app_version: 1.6.1
+  version: 0.1.1
```
Comment (ananth102) [inline tools/helmify/src/config.yaml:77]: Why is this a patch version increase, when istio is a minor version increase?

File: tools/helmify/src/config.yaml [hunk 2]
```diff
@@ -44,14 +44,14 @@ katib:
...
-    - upstream/common/istio-1-14/istio-crds/base
-    - upstream/common/istio-1-14/istio-namespace/base
-    - upstream/common/istio-1-14/istio-install/base
-  output_helm_chart_path: charts/common/istio-1-14
-  version: 0.1.1
-  app_version: 1.14.1
+    - upstream/common/istio-1-16/istio-crds/base
+    - upstream/common/istio-1-16/istio-namespace/base
+    - upstream/common/istio-1-16/istio-install/base
+  output_helm_chart_path: charts/common/istio-1-16
```
Comment (ananth102) [inline tools/helmify/src/config.yaml:52]: didnt the folder get changed to istio? Ignore if the folder name change is temporary.

File: tools/helmify/src/config.yaml (hunk 1 above)
Comment (jsitu777) [reply tools/helmify/src/config.yaml]: good catch, addressed.

File: tools/helmify/src/config.yaml (hunk 2 above)
Comment (jsitu777) [reply tools/helmify/src/config.yaml]: good catch, addressed.

**Review by ryansteakley — APPROVED** [review]

**Pull request merged**